### DISCUSSION TITLE
Getting started. What is actually needed?

### DISCUSSION THREAD

**Discussion by unitrix0** [AvaloniaUI/Avalonia#17793] [Q&A] [answered]
So, I've just made my first steps with Avalonia or I tried I should say. I've added the templates with ```dotnet new install Avalonia.Templates``` and I've installed the Rider extension. Nevertheless when I create a new Avalonia Solution or open the examples, I see a lot of red squiggles. Is there anything else I need to Install? 
![grafik](https://github.com/user-attachments/assets/8c520f72-2754-46c3-8777-a3f3119a1166)

**Comment by thevortexcloud** [comment]
You may need to build the app once/run a `dotnet restore`. You also need to install various dotnet workloads for specific platforms, EG android and Browser. Desktop however should "just work" assuming all the packages are downloaded.

**Reply by unitrix0** [reply]
Ok that helped me out a bit. ```dotnet restore``` told me to run ```dotnet workload restore``` which restored a couple of dependencies. After that I ran another ```dotnet restore``` and no everything except the Android project is without errors. The build error tells me to install the [Android SDK](https://aka.ms/dotnet-android-install-sdk).

How is this all necessary but not mentioned in the installation instructions? I don't think that it is usual that an Android SDK is installed by default.

**Reply by unitrix0** [reply]
so looks like I need to install Xamarin too? I'm lost...

**Reply by thevortexcloud** [reply]
It's only needed for android/a couple other platforms. If you are just targeting desktop then you don't need any of the workloads and you can delete the platforms you are not targeting from the solution. The only requirement to build for desktop is the nuget packages.

>  I don't think that it is usual that an Android SDK is installed by default.

It's not. But it is required to build .NET apps for android. I think Visual Studio can install it for you in the installer on Windows though.

>so looks like I need to install Xamarin too? I'm lost...

You don't need Xamarin. When you target iOS/Android in .NET it automatically adds implicit references to Xamarin libraries. It's sort of explained here:

https://learn.microsoft.com/en-us/dotnet/standard/frameworks

**Reply by unitrix0** [reply]
Ok, so I've installed JDK 17 and I just ended up at the next error. It says that API Level 34 isn't installed and it could be fixed by executing ```"dotnet build -t:InstallAndroidDependencies -f" net8.0-android "-p:AndroidSdkDirectory=C:\Program Files (x86)\Android\android-sdk"```. Which then leads me to the next error stating that ```AvaloniaApplication1\AvaloniaApplication1.Desktop\obj\project.assets.json``` has no target for ```net8.0-android``` and that the Android SDK license is  not accepted. This should be fixed by executing ```sdkmanager.bat --licenses``` which of course leads to an exception. 

I'm giving up on this.

**Reply by thevortexcloud** [reply]
Getting android setup for .NET is rather complicated. It's a lot simpler if you use Rider though. As Idea is the official android IDE, so Jetbrains has a lot of built in tools for setting it up and managing it. But I don't blame you for giving up on it. It's something Microsoft/Google really needs to work on making simpler.

**Reply by unitrix0** [reply]
> It's something Microsoft/Google really needs to work on making simpler.

Seems to be true yes.

So I started all over and eventually managed that I could build without errors. However, if I try to run it in the emulator it somehow finds a Java SDK 1.8.0 somehwere and complains that it needs at least 11.0. I have not the slightest idea where this is defined or found! Maybe someone elese can give me a hint?

**Reply by thevortexcloud** [reply]
From memory, most programs check the `JAVA_HOME` environment variable to find where Java is. Not sure if the android emulator does though.

**Reply by unitrix0** [reply]
Yes, I also came across that variable. It wasn't set, so I did it manually, but it didn't help.

**Reply by unitrix0** [reply, marked as answer]
I've fixed it.

Rider has its own setting for the SDK. It's under ```File | Settings | Build, Execution, Deployment | Android```
God knows why it installs such an old SDK.
